Simplified/Reader2/LCP/LCPAuthenticationViewController.swift

```diff
@@ -0,0 +1,214 @@
+//
+//  LCPAuthenticationViewController.swift
+//  r2-testapp-swift
+//
+//  Created by Mickaël Menu on 01.03.19.
+//
+//  Copyright 2019 Readium Foundation. All rights reserved.
+//  Use of this source code is governed by a BSD-style license which is detailed
+//  in the LICENSE file present in the project repository where this source code is maintained.
+//
+
+#if LCP
+
+import SafariServices
+import UIKit
+import ReadiumLCP
+
+
+protocol LCPAuthenticationDelegate: AnyObject {
+    
+    func authenticate(_ license: LCPAuthenticatedLicense, with passphrase: String)
+    func didCancelAuthentication(of license: LCPAuthenticatedLicense)
+
+}
+
+class LCPAuthenticationViewController: UIViewController {
+    
+    weak var delegate: LCPAuthenticationDelegate?
+    
+    @IBOutlet weak var scrollView: UIScrollView!
+    @IBOutlet weak var hintLabel: UILabel!
+    @IBOutlet weak var promptLabel: UILabel!
+    @IBOutlet weak var messageLabel: UILabel!
+    @IBOutlet weak var passphraseField: UITextField!
+    @IBOutlet weak var supportButton: UIButton!
+    
+    private let license: LCPAuthenticatedLicense
+    private let reason: LCPAuthenticationReason
+    private let supportLinks: [(Link, URL)]
+    
+    init(license: LCPAuthenticatedLicense, reason: LCPAuthenticationReason) {
+        self.license = license
+        self.reason = reason
+        self.supportLinks = license.supportLinks
+            .compactMap { link -> (Link, URL)? in
+                guard let url = URL(string: link.href), UIApplication.shared.canOpenURL(url) else {
+                    return nil
+                }
+                return (link, url)
+            }
+
+        super.init(nibName: nil, bundle: nil)
+
+        NotificationCenter.default.addObserver(self, selector: #selector(keyboardWillChangeFrame(_:)), name: UIResponder.keyboardWillChangeFrameNotification, object: nil)
+        NotificationCenter.default.addObserver(self, selector: #selector(keyboardWillChangeFrame(_:)), name: UIResponder.keyboardWillHideNotification, object: nil)
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    deinit {
+        NotificationCenter.default.removeObserver(self)
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        var provider = license.document.provider
```

Comment:
for example the way you derive the provider is not really view logic.

That way we could add some unit tests which I think would be good.

Reply:
May be I don't understand what you mean here exactly, license and document contain parsed .lcpl data. Parsing is done in r2-lcp-swift, and both objects are defined there as well. It is possible to add unit tests for LicenseDocument and LCPAuthenticatedLicense, they are not covered in r2-lcp-swift.
`provider` here is just a string, contains a name or a URL address of the license provider, and is displayed on the view, among other information about the license.

Reply:
what I mean is that the VC shouldn't need to deal with the license and its properties. If the license class changes, then this VC will need to change too, and not for a reason related to view handling. Can we add a `provider` computed property to the License class that returns this String? Or have the delegate do that?

Reply:
LCPAuthenticatedLicense is in r2-lcp-swift, probably it's better to avoid changes as we don't really need to change that library. Added an adapter class LCPLicenseInfo for that.

Reply:
I agree to not have tons of changes in our fork, I meant to just have a computed property as an extension. That could live anywhere. But in any case I'll take a look at the adapter, it's probably fine.